tiledb/sm/rest/curl.cc

```diff
@@ -633,10 +639,17 @@ Status Curl::make_curl_request_common(
       }
     }
 
+    bool should_retry = should_retry_request(curl_code, http_code);
+    // Handle the error if we shouldn't retry and the curl request failed.
+    if (!should_retry && curl_code != CURLE_OK) {
+      return LOG_STATUS(Status_RestError(
+          std::string("Error submitting the Curl request: ") +
+          get_curl_errstr(curl_code)));
+    }
```

Comment:
Wasn't this already handled in `check_curl_errors_v2` [here](https://github.com/TileDB-Inc/TileDB/blob/dec249a9e3350402ae6703075047960b43c35f05/tiledb/sm/rest/curl.cc#L1083) for example for `get_data` and in `check_curl_errors` for the other methods? Why this didn't work for this case?

I am a bit scared to touch this logic as it has been the source of defects multiple times in the past due to the way return codes and errors are handled, so I want to be 100% sure we are doing the right fix here.